src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketPal.Unix.cs

```diff
@@ -800,12 +796,13 @@ public static unsafe bool TryCompleteReceive(SafeSocketHandle socket, Span<byte>
             }
         }
 
-        public static unsafe bool TryCompleteReceiveFrom(SafeSocketHandle socket, Span<byte> buffer, IList<ArraySegment<byte>>? buffers, SocketFlags flags, byte[]? socketAddress, ref int socketAddressLen, out int bytesReceived, out SocketFlags receivedFlags, out SocketError errorCode)
+        public static unsafe bool TryCompleteReceiveFrom(SafeSocketHandle socket, Span<byte> buffer, IList<ArraySegment<byte>>? buffers, SocketFlags flags, Span<byte> socketAddress, out int socketAddressLen, out int bytesReceived, out SocketFlags receivedFlags, out SocketError errorCode)
         {
             try
             {
                 Interop.Error errno;
                 int received;
+                int socketAddressLength = 0;
```

Comment:
It's confusing to me reading the code that we have both socketAddressLen and now socketAddressLength. Can we name this something else, like receivedSocketAddressLength, to make it clearer what it is and if nothing else to more visually separate it from the other?

Reply:
updated & simplified.